src/WebJobs.Extensions.MobileApps/Bindings/MobileTableQueryValueProvider.cs

```diff
@@ -2,26 +2,25 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Host.Bindings;
 using Microsoft.WindowsAzure.MobileServices;
 
 namespace Microsoft.Azure.WebJobs.Extensions.MobileApps
 {
-    internal class MobileTableQueryValueProvider<T> : IValueProvider
+    internal class MobileTableQueryValueProvider<T> : IValueBinder
```

Comment:
There's no helpers for dealing with IValueProviders -- I'm not sure which is better: having both IValueProvider and IValueBinder or just having IValueBInder but making SetAsync a no-op. I'd like us to narrow down the interfaces we tell people to deal with :-)

I can't remove this class (yet) b/c BindToExactType won't work due to the open generics. Functions can have IMobileServiceTable<T> as a parameter and I want all of those to run the GetValue() code below.

---
In reply to: [63598844](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/97#discussion_r63598844) [](ancestors = 63598844)